### PR TITLE
Sharing `content_for` theme checks with `render` theme checks

### PR DESCRIPTION
**Pull request by aswamy**
## What are you adding in this PR?

Closes https://github.com/Shopify/developer-tools-team/issues/674

Add `content_for` theme checks + refactor a lot of tests to ensure we don't repeat all the whitespace/coma stuff across multiple tests.

This PR HAS to be reviewed by commits or it doesn't make sense:

- Commit 1: rename `render` tag theme checks so it says "argument" instead of "parameters"
  -  you don't really have to be thorough with this since no logic is changed. public docs will be updated soon
- Commit 2: refactoring `render` tag theme check logic into separate files
  - mostly just moved things into separate files
- Commit 3: `content_for` theme checks
  - the tests are mostly copied from `render` tag theme checks so nothing special here
  - some shared test cases have been moved to `arguments.spec.ts`, but a lot is still shared with `render` tag. Didn't want to create another shared file that tests both tags because of readability + issues around `alias` argument for `render` tag

## What's next? Any followup issues?

more content_for tag support

## Tophat

- Create a theme with a block with the following liquid doc
```
{% doc %}
  @param {product} [my_prod]
  @param {string} my_string
  @param {number} my_number
{% enddoc %}
```

- In another liquid file, render the `content_for` tag and ensure all 4 theme checks are working
  - missing content_for arg: `{% content_for 'block', type: 'swatches', id: 'fake' %}`
  - duplicate content_for arg: `{% content_for 'block', type: 'swatches', id: 'fake', my_string: '', my_string: '' %}`
  - unrecognized content_for arg: `{% content_for 'block', type: 'swatches', id: 'fake', foo: 123 %}`
  - valid content_for arg type: `{% content_for 'block', type: 'swatches', id: 'fake', my_string: 123 %}`

- try the `suggestion` fix from VSCode

## Before you deploy
- [x] I included a minor bump `changeset`
- [x] My feature is backward compatible

### CODE REVIEW

File: packages/theme-check-common/src/checks/valid-render-snippet-argument-types/index.ts [hunk 1]
```diff
@@ -10,15 +10,15 @@ import {
 } from '../../liquid-doc/utils';
 import { StringCorrector } from '../../fixes';
 
-export const ValidRenderSnippetParamTypes: LiquidCheckDefinition = {
+export const ValidRenderSnippetArgumentTypes: LiquidCheckDefinition = {
   meta: {
-    code: 'ValidRenderSnippetParamTypes',
-    name: 'Valid Render Snippet Parameter Types',
+    code: 'ValidRenderSnippetArgumentTypes',
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-check-common/src/checks/valid-render-snippet-argument-types/index.ts:15]: e.g. https://github.com/Shopify/theme-tools/blob/main/packages/theme-check-common/src/checks/valid-json/index.ts#L6
https://github.com/Shopify/theme-tools/blob/main/packages/theme-check-common/src/checks/liquid-html-syntax-error/index.ts#L24

File: packages/theme-check-common/src/checks/index.ts [hunk 2]
```diff
@@ -99,25 +106,26 @@ export const allChecks: (LiquidCheckDefinition | JSONCheckDefinition)[] = [
...
+  ValidRenderSnippetArgumentTypes,
   ValidSchema,
   ValidSettingsKey,
   ValidStaticBlockType,
   ValidVisibleIf,
   ValidVisibleIfSettingsSchema,
   VariableName,
-  MissingRenderSnippetArguments,
-  UnrecognizedRenderSnippetArguments,
-  ValidRenderSnippetArgumentTypes,
```
Comment (charlespwd) [inline packages/theme-check-common/src/checks/index.ts]: Are they all moved or should they be aliased?

**Review by charlespwd — APPROVED** [review]
I'm not sure anymore we need an alias. But if there is an oldname that we no longer support, we should make it so that the previous configuration somehow migrates somewhere. Kind of a nit. Might not be a huge deal.

 But I know I don't like upgrading something and getting config errors :P

**Comment by aswamy** [comment]
@charlespwd since those theme checks are not released yet, should we _do_ aliases?

I created a theme using `shopify theme init` (that does have those theme checks), but does not create them in the `.theme-check.yml` file.
